src/source/vector_tile_worker_source.js

```diff
@@ -203,13 +207,27 @@ class VectorTileWorkerSource implements WorkerSource {
             // response.vectorTile will be present in the GeoJSON worker case (which inherits from this class)
             // because we stub the vector tile interface around JSON data instead of parsing it directly
             workerTile.vectorTile = response.vectorTile || new vt.VectorTile(new Protobuf(rawTileData));
+            const parseTile = () => {
+                workerTile.parse(workerTile.vectorTile, this.layerIndex, this.availableImages, this.actor, (err, result) => {
+                    if (err || !result) return callback(err);
+    
+                    // Transferring a copy of rawTileData because the worker needs to retain its copy.
+                    callback(null, extend({rawTileData: rawTileData.slice(0)}, result, cacheControl, resourceTiming));
+                });
+            };
 
-            workerTile.parse(workerTile.vectorTile, this.layerIndex, this.availableImages, this.actor, (err, result) => {
-                if (err || !result) return callback(err);
-
-                // Transferring a copy of rawTileData because the worker needs to retain its copy.
-                callback(null, extend({rawTileData: rawTileData.slice(0)}, result, cacheControl, resourceTiming));
-            });
+            if (this.isSpriteLoaded) {
+                parseTile();
+            } else {
+                const metadata = {type: 'parseTile', isSymbolTile: params.isSymbolTile, zoom: params.tileZoom};
+                this.once('isSpriteLoaded', () => {
+                    if (this.scheduler) {
+                        this.scheduler.add(parseTile, metadata);
+                    } else {
+                        parseTile();
```

Comment:
In what case is `this.scheduler` not defined?

Reply:
I copied this from above in the `DedupedRequest` class. Both classes get `scheduler` from `actor.scheduler` and since `DedupedRequest` lists the type as `?Scheduler`, I assumed that should apply to this class as well and I should guard against the possibility that `actor` won't exist or won't have a `scheduler` property. I think this could be an issue in testing?